v8_c_api/src/v8_c_api.cpp

```diff
@@ -306,6 +307,303 @@ v8_pd_list* v8_PDListCreate(v8::ArrayBuffer::Allocator *alloc) {
     return native_data;
 }
 
+// Some parts of the contents of this anonymous namespace below
+// were borrowed and changed.
+namespace {
+/*
+MIT License
+
+Copyright (c) 2019 Elmi Ahmadov
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*/
+
+static inline v8_inspector::StringView convertToStringView(const std::string &str) {
+    auto* stringView = reinterpret_cast<const uint8_t*>(str.c_str());
+    return { stringView, str.length() };
+}
+
+static inline std::string convertToString(v8::Isolate* isolate, const v8_inspector::StringView stringView) {
+    int length = static_cast<int>(stringView.length());
+    v8::Local<v8::String> message = (
+        stringView.is8Bit()
+          ? v8::String::NewFromOneByte(isolate, reinterpret_cast<const uint8_t*>(stringView.characters8()), v8::NewStringType::kNormal, length)
+          : v8::String::NewFromTwoByte(isolate, reinterpret_cast<const uint16_t*>(stringView.characters16()), v8::NewStringType::kNormal, length)
+      ).ToLocalChecked();
+    v8::String::Utf8Value result(isolate, message);
+    return *result;
+}
+
+class v8_inspector_channel_wrapper final: public v8_inspector::V8Inspector::Channel {
+public:
+    explicit v8_inspector_channel_wrapper(
+        v8::Isolate *isolate,
+        const std::function<void(std::string)> &onResponse = {}
+    );
+
+    void sendResponse(int callId, std::unique_ptr<v8_inspector::StringBuffer> message) override;
+    void sendNotification(std::unique_ptr<v8_inspector::StringBuffer> message) override;
+    void flushProtocolNotifications() override;
+
+    void setIsolate(v8::Isolate *isolate);
+
+    void setOnResponseCallback(const std::function<void(std::string)> &callback);
+
+private:
+    v8::Isolate* isolate_;
```

Comment:
I see that you need this isolate pointer only to convert from string view to string? If this is the case, can we find another way to do the conversion without the isolate?

Reply:
The string view has 2 methods, `characters8`, `characters16`. And we can know which one to use using `is8Bit`. So its either convert from utf8 and utf16 no?